### PR TITLE
LGFS transfer fee calculation

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Transfer fee calculation

#### Ticket

[CBO-492](https://dsdmoj.atlassian.net/browse/CBO-492)

#### Why
So users do not have to use spreadsheets to work out the transfer fee amount
to claim.

#### How
Extend GraduatedPrice logic to include transfer fee
bill scenario mappings

--------

#### TODO (wip)

 - [x] Front end design improvements
    - [x] form page
    - [x] summary page
    - [x] check your claim page
    - [x] submitted claim summary page/case worker side
 - [x] apply calc JS
 - [x] extend/amend backend **
 - [x] Add/amend specs (+/-PriceController spec,+/- GraduatedPrice spec)
 - [x] Add/amend features +/-Transfer claim feature, +fee calc transfer claim feature
 - [x] refactor

#### Extensions **
 - [x] add :days attribute to fee type/or use actual trial length
 - [x] add days_claimable? flag to transfer details and conditionally display days/actual_trial_length attribute (should i use retrial_actual_length for retrial case conclusions??)
 - [x] Add transfer detail summary to transfer fee page??

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -1014,13 +1021,12 @@ en:
...
           page_hint: *fees_vat_prompt
           actual_trial_length: Actual trial length
-          graduated_fees: Graduated fees
-          graduated_prompt_text: *graduated_fee_prompt_content
+          actual_trial_length_hint: Number of days
           rate: *rate
-          quantity: Total pages of evidence
-          quantity_hint: Enter a quantity
+          quantity: *ppe_quantity
+          quantity_hint: Number of pages
```
Comment (jsugarman) [inline config/locales/en.yml]: remove

File: features/page_objects/sections/lgfs_transfer_detail_section.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-class TransferFeeSection < SitePrism::Section
+class LgfsTransferDetailSection < SitePrism::Section
   include SelectHelper
 
   element :transfer_fee_total, "#claim_transfer_fee_attributes_amount"
```
Comment (jsugarman) [inline features/page_objects/sections/lgfs_transfer_detail_section.rb]: remove

File: features/step_definitions/litigator_transfer_claim_steps.rb [hunk 3]
```diff
@@ -29,6 +29,23 @@
...
+  end
+end
+
+Then(/^I should (not )?see the ppe field$/) do |negate|
+  if negate
+    expect(@litigator_transfer_claim_form_page.transfer_fee).to_not have_ppe_total
+  else
+    expect(@litigator_transfer_claim_form_page.transfer_fee).to have_ppe_total
+  end
+
```
Comment (jsugarman) [inline features/step_definitions/litigator_transfer_claim_steps.rb]: remove

File: spec/models/claim/transfer_brain_data_item_collection_spec.rb [hunk 4]
```diff
@@ -159,6 +159,26 @@
...
+
+      it 'returns for matching detail' do
+        is_expected.to be_in(%w[TRUE FALSE])
+      end
+    end
+
+    context 'when given valid details with a wildcard case conclusion id' do
+      let(:detail) { with_wildcard_mapping }
+
+      it 'returns bill scenario for matching detail' do
```
Comment (jsugarman) [inline spec/models/claim/transfer_brain_data_item_collection_spec.rb]: correct description

File: spec/models/claim/transfer_brain_spec.rb [hunk 5]
```diff
@@ -93,12 +93,19 @@
...
 
   describe '.ppe_required' do
-    it 'returns a boolean string denoting wether PPE quantity is required for the transfer fee' do
+    it 'returns a boolean string denoting whether PPE quantity is required for the transfer fee' do
       td = transfer_detail('new', true, 10)
       expect(described_class.ppe_required(td)).to eq "FALSE"
     end
   end
 
+  describe '.day_required' do
```
Comment (jsugarman) [inline spec/models/claim/transfer_brain_spec.rb]: correct description

**Review by colinbruce — APPROVED** [review]
LGTM 👍

**Pull request merged**